### PR TITLE
Add an option to force trainer pokemons to be fully evolved after a given level

### PR DESCRIPTION
**Pull request by Eijebong**
This was tested during an archipelago with 4 emerald players with this option on.

This was tested with a patch on your emerald-stable branch though, so hopefully I didn't forget anything when migrating it to target the dev branch.

### CODE REVIEW

File: worlds/pokemon_emerald/__init__.py [hunk 1]
```diff
@@ -848,13 +849,15 @@ def randomize_opponent_parties() -> None:
...
                     target_type = self.random.choice(original_species.types) if should_match_type else None
+                    force_fully_evolved = force_fully_evolved_at != 0 and pokemon.level >= force_fully_evolved_at
 
                     new_species = get_random_species(
                         self.random,
                         self.modified_species,
                         target_bst,
                         target_type,
-                        LEGENDARY_POKEMON if allow_legendaries else set()
+                        LEGENDARY_POKEMON if allow_legendaries else set(),
```
Comment (Eijebong) [inline worlds/pokemon_emerald/__init__.py:859]: Also while porting my patch to the dev branch I saw this which looks reversed? This is a blacklist, which blacklists legendary pokemons when `allow_legendary` is on. Although this is out of scope for this PR, thought I'd let you know :)

**Comment by Zunawe** [comment]
I appreciate the contribution, but unfortunately I just happened to make this myself a couple weeks ago just before I left for vacation and didn't push it.

I'll still use this to ask why you set 40 as the upper limit. Was it arbitrary, historical, informed, etc..? Do you object to my choice to let it go higher? Your limit would catch certain vanilla evolutions (like pseudo-legendaries) early.

File: worlds/pokemon_emerald/__init__.py (hunk 1 above)
Comment (Zunawe) [reply worlds/pokemon_emerald/__init__.py:859]: Not only did I make this mistake here, I made it a second time when refactoring to add this option, lol. So thanks for catching that.

**Comment by Eijebong** [comment]
Ha welp :) I'll close this then. I set 40 as the upper limit arbitrarily and then forgot to think about it again
